src/Symfony/Component/Config/Tests/Builder/Fixtures/NodeInitialValues/Symfony/Config/NodeInitialValuesConfig.php

```diff
@@ -28,9 +41,22 @@ public function someCleverName(array $value = []): \Symfony\Config\NodeInitialVa
         return $this->someCleverName;
     }
 
-    public function messenger(array $value = []): \Symfony\Config\NodeInitialValues\MessengerConfig
+    /**
+     * @template TValue
+     * @param TValue $value
+     * @return \Symfony\Config\NodeInitialValues\MessengerConfig|$this
+     * @psalm-return (TValue is array ? \Symfony\Config\NodeInitialValues\MessengerConfig : static)
```

Comment:
why should passing an array return a MessengerConfig while passing another kind of value would be fluent ? What does this correspond to ?

Reply:
I wondered about the same and I think I've got the answer:
- When you pass a non-array, you likely won't want to configure anything else nested in the tree you're configuring. That's what happens for other config formats, structurally.
- When you pass an array, you can configure nested options, so that then it might make sense to return the nested config-builder.

I'm just wondering about the check: should we return the config-builder when the array is non empty?
Or should we do this?

```suggestion
     * @psalm-return (TValue is non-empty-array ? $this :\Symfony\Config\NodeInitialValues\MessengerConfig)
```